grpc-client-server/prober_client/main.go

```diff
@@ -26,13 +62,73 @@ func main() {
 	c := pb.NewProberClient(conn)
 
 	// Contact the server and print out its response.
-	ctx := context.Background() // TODO: add a timeout
+	var wg sync.WaitGroup
+	for _, endpoint := range endpoints {
+		req := &pb.ProbeRequest{Endpoint: endpoint, NumberOfRequests: int32(*nOfRequests)}
+		wg.Add(1)
+		go SingleProbe(os.Stdout, c, req, &wg)
+	}
+	wg.Wait()
+}
+
+func SingleProbe(w io.Writer, c pb.ProberClient, req *pb.ProbeRequest, wg *sync.WaitGroup) {
+	results := make(chan *Result)
+	timeout := time.Duration(*timeout) * time.Second
+	go CreateProgressBar(w, timeout, req.Endpoint, results, wg)
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	r, err := c.DoProbes(ctx, req)
+	cancel()
 
-	// TODO: endpoint should be a flag
-	// TODO: add number of times to probe
-	r, err := c.DoProbes(ctx, &pb.ProbeRequest{Endpoint: "http://www.google.com"})
 	if err != nil {
-		log.Fatalf("could not probe: %v", err)
+		results <- &Result{Endpoint: req.Endpoint, Err: err}
+		return
+	}
+
+	results <- &Result{
+		Endpoint: req.Endpoint,
+		Failed:   r.FailedRequests,
+		Average:  r.AverageResponseTime,
+	}
+}
+
+func CreateProgressBar(w io.Writer, timeout time.Duration, endpoint string, results <-chan *Result, wg *sync.WaitGroup) {
+	defer wg.Done()
+	ticker := 200 * time.Millisecond
```

Comment:
If I set my timeout to 1s, what will I see the progress bar do?

Reply:
It's not ideal at the moment, it doesn't show the progress of requests but only the progress of timeout. 

So if you provide 1s timeout it will have 5 `ticks` (20%, 40%, etc). If the request completes in 0.5s then it will just fill up to 100%. 

In that sense, I guess it's more of a countdown than a progress bar :D

Reply:
Will the first tick _actually_ hit 20% though? How does the progress bar know that 1 tick is 20%?